broker-daemon/state-machines/fill-state-machine.js

```diff
@@ -406,6 +424,30 @@ const FillStateMachine = StateMachine.factory({
   }
 })
 
+/**
+ * serialize an fill for transmission via grpc
+ * @param {Object} fillObject - Plain object representation of the fill, state, dates
+ * @returns {Object} Object to be serialized into a GRPC message
+ */
+FillStateMachine.serialize = function (fillObject) {
+  const {
+    fill,
+    state,
+    error,
+    dates
+  } = fillObject
+
+  const serializedFill = fill.serialize()
+
+  Object.assign(serializedFill, {
```

Comment:
[nit] generally you don't want to `Object.assign` to an object that you didn't create as you may unintentionally mutate something. Two ways to do this that would avoid that issue:

```
Object.assign({}, serializedFill, { other stuff })
```

or

```
return {
  fillStatus: state.toUpperCase(),
  error: error ? error.toString() : undefined,
  dates,
  ...serializedFill
}
```